assignments/prototype-refactor.js

```diff
@@ -7,3 +7,133 @@ Prototype Refactor
 2. Your goal is to refactor all of this code to use ES6 Classes. The console.log() statements should still return what is expected of them.
 
 */
+
+
+class GameObject {
+	constructor(thing) {
+	this.createdAt = thing.createdAt;
+	this.name = thing.name;
+	this.dimensions = thing.dimensions;
+		}
+	destroy() {
+		return `${this.name} was removed from the game in the most unpleasant way.`
+	}
+	
+}
+
+//  === CharacterStats ===
+//  * healthPoints
+//  * takeDamage() // prototype method -> returns the string '<object name> took damage.'
+//  * should inherit destroy() from GameObject's prototype
+
+class CharacterStats extends GameObject {
+	constructor(stats){
+	super(stats)
+	this.healthPoints = stats.healthPoints;
+	}
+	takeDamage () {
+		return `${this.name} took damage! But was it super effective?`
+}
+	
+}
+
+
+/*
+  === Humanoid (Having an appearance or character resembling that of a human.) ===
+  * team
+  * weapons
+  * language
+  * greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
+  * should inherit destroy() from GameObject through CharacterStats
+  * should inherit takeDamage() from CharacterStats
+*/
+ class Humanoid extends CharacterStats {
+	 constructor(attr) {
+	 
+	 super(attr)
+	 this.team = attr.team;
+	 this.weapons = attr.weapons;
+	 this.language = attr.language
+		 }
+	 greet () {
+		 return `${this.name} offers a cold greeting in ${this.language}`
```

Comment:
We've been having waaay too much gritty fantasy worlds in the media recently. We should have one with rainbows, and warm greetings, and fuzzy what-nots!